.travis.yml

```diff
@@ -197,10 +197,19 @@ jobs:
 
   - stage: checks 🔬
     if: (type != pull_request AND branch = master) OR (tag =~ ^v)
-    name: "Tests"
+    name: "Tests (http-bridge)"
     script:
     - travis_retry curl -L https://github.com/KtorZ/cardano-http-bridge/releases/download/v0.0.5/hermes-testnet.tar.gz | tar xz -C $HOME
 
+    - stack --no-terminal test --fast --haddock --no-haddock-deps --coverage cardano-wallet-http-bridge --ta "--skip MERGE_DISABLED"
+    - mkdir -p .coverage/http-bridge && find . -name "*.tix" ! -path "*.coverage*" -exec cp \{} .coverage/http-bridge \;
+
+    - tar czf $STACK_WORK_CACHE .stack-work .coverage lib/**/.stack-work lib/**/*.tix
```

Comment:
I'm not sure if these stages write to the cache or not.

If the `Tests` would overwrite the cache with a `.stack-work` without cache for `cardano-http-bridge`, that would likely make PR builds depending on `cardano-http-bridge` take too long. I'm not sure this is what will happen though.

But we won't know if this works until we merge to master and create/merge? another PR.

Reply:
Hm, not sure how the caching works between jobs. However I see that:

```
# Caching so the next build will be fast too:
cache:
  timeout: 1337
  directories:
  - $HOME/.stack
```

and

```
STACK_WORK_CACHE=$HOME/.stack/stack-work.tar.gz
```

and there are calls like `tar czf $STACK_WORK_CACHE .stack-work...` on different jobs (e.g. on "Compiling project" and "Tests").
Does it mean that `$STACK_WORK_CACHE` is overritten by subsequent jobs, or is it somehow merged? If the latter, then all should be fine, I suppose... :man_shrugging:

Reply:
So `- stage`s are run sequentially, and jobs are run in parallel. 

I _do_ know that jobs function like this regarding caching
```
c <- getCache
work <- sleep 1h
putCache (c : work)
```
because we previously had problems with race conditions from parallel jobs on the CI because of this. **I.e. the caches are not merged.**

What I don't get though is why `tar czf $STACK_WORK_CACHE .stack-work` is not only called in `cache` stage, but also in the `test` stage

Reply:
Actually… then maybe its fine?